### ISSUE TITLE
Jekyll 3.9.0 test failure with ruby-rouge 3.23.0

### ISSUE THREAD

**Issue by dleidert** [jekyll/jekyll#8436]
## My Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Operating System | Debian           |
| `jekyll`         | 3.9.0     |
| `github-pages`   | none     |
| `ruby-rouge` | 3.25.0 |
---

## Expected Behaviour

Up to ruby-rouge 3.21.0 the tests succeeded as expected. With the latest release one test fails expecting a different output.

## Current Behavior

```
Failure:
Minitest::Result#test_: with the rouge highlighter post content has highlight tag with linenumbers should should stop highlighting at boundary with rouge 2.  [/<<PKGBUILDDIR>>/test/test_tags.rb:482]
Minitest::Assertion: Expected /<p>This\ is\ not\ yet\ highlighted<\/p>\n\n<figure\ class="highlight"><pre><code\ class="language\-php"\ data\-lang="php"><table\ class="rouge\-table"><tbody><tr><td\ class="gutter
\ gl"><pre\ class="lineno">1\n<\/pre><\/td><td\ class="code"><pre><span\ class="nx">test<\/span>\n<\/pre><\/td><\/tr><\/tbody><\/table><\/code><\/pre><\/figure>\n\n<p>This\ should\ not\ be\ highlighted,\ right\?
<\/p>\n/ to match "<hr />\n<p>title: This is a test\n—</p>\n\n<p>This is not yet highlighted</p>\n\n<figure class=\"highlight\"><pre><code class=\"language-php\" data-lang=\"php\"><table class=\"rouge-table\"><t
body><tr><td class=\"gutter gl\"><pre class=\"lineno\">1\n</pre></td><td class=\"code\"><pre><span class=\"n\">test</span>\n</pre></td></tr></tbody></table></code></pre></figure>\n\n<p>This should not be highlig
hted, right?</p>\n".
```
Note the `class="nx"` vs `class="n"` output.

I reported the issue to the rouge project too rouge-ruby/rouge#1603. But it seems it is considered "updated behavior". Can you please check?

**Comment by ashmaroli** [comment]
Hello @dleidert 
Let me guess, you were building a package using v3.9.0 and its unit tests messes up with your template... ?

Jokes apart, are you willing to submit a pull request against the `3.9-stable` branch?
The `master` already has this patched via https://github.com/jekyll/jekyll/commit/9a7955d9b3e85a78fa4876ccd98708fb2f762472

**Comment by dleidert** [comment]
Nope. Before we upload a new version of a package (here rouge 3.23) we rebuild all packages build-depending on it (e.g. kramdown and jekyll) and also run all their tests. So we can spot build issues or regressions before our users get the update. I think it is very similar to what you do to make sure nothing breaks.

Regarding the quick fix. Wouldn't it be more appropriate for you as upstream to either use a regular expression to compare the output or expect different results depending on the rouge version so the test succeeds with all supported rouge versions? I'm willing to provide a patch. It will just need a few days because I'm very short on free time.

**Comment by ashmaroli** [comment]
> Wouldn't it be more appropriate for you as upstream to either use a regular expression to compare the output or expect different results depending on the rouge version

The `3.9-stable` branch tests against across major versions of Rouge to maintain compatibility.
However, the `master` version supports only Rouge 3.x. So we test against just the latest Rouge version on `master`.

We don't intend to test against each and every supported version of a runtime-dependency.. just the latest version in a given major-version-bracket.

**Comment by dleidert** [comment]
Ok. I have prepared a relaxed test for the 3.9 branch I apply in the Debian package too. It also contains a relaxation for the changes made in the latest Pygments package (fails another test).

**Comment by ashmaroli** [comment]
Pull request welcome.
Will comment further on seeing the code / CI results..
